src/compilation/scope_manager.rs

```diff
@@ -46,21 +122,37 @@ impl ScopeManager {
         self.funcs.pop();
     }
 
+    /// Searches through all scopes (starting with the innermost scope) for a variable named `name`.
     pub fn get_var(&self, name: &str) -> Option<&Var> {
         self.vars.iter().rev().find_map(|s| s.get(name))
     }
 
+    /// Sets a variable named `name` in the current scope.
     pub fn set_var(&mut self, name: &str, var_type: Type, value: LLVMValueRef) {
         let cur_scope = self.vars.last_mut().unwrap();
         cur_scope.insert(name.to_string(), Var { var_type, value });
     }
 
+    // TODO: Quick fix: if LLVMValueRef hashes then we should use that for lookup.
+    //  Take a look at get_func() usage and see how hard we work to convert ValueRef to name.
```

Comment:
Oh I love this. Another easy thing that is super cool to implement!

Reply:
yay!